crates/astria-sequencer/src/slinky/oracle/state_ext.rs

```diff
@@ -0,0 +1,289 @@
+
+            let currency_pair = key
+                .strip_prefix(&prefix)
+                .context("failed to strip prefix from currency pair state key")?
+                .parse::<CurrencyPair>()
+                .context("failed to parse storage key suffix as currency pair")?;
+            currency_pairs.insert(id, currency_pair);
+        }
+        Ok(currency_pairs)
+    }
+
+    #[instrument(skip(self))]
+    async fn get_num_currency_pairs(&self) -> Result<u64> {
+        let Some(bytes) = self
+            .get_raw(NUM_CURRENCY_PAIRS_KEY)
+            .await
+            .context("failed reading number of currency pairs from state")?
+        else {
+            return Ok(0);
+        };
+        let Count(num_currency_pairs) =
+            Count::try_from_slice(&bytes).context("invalid number of currency pairs bytes")?;
+        Ok(num_currency_pairs)
+    }
+
+    #[instrument(skip(self))]
+    async fn get_num_removed_currency_pairs(&self) -> Result<u64> {
+        let Some(bytes) = self
+            .get_raw(NUM_REMOVED_CURRENCY_PAIRS_KEY)
+            .await
+            .context("failed reading number of removed currency pairs from state")?
+        else {
+            return Ok(0);
+        };
+        let Count(num_removed_currency_pairs) = Count::try_from_slice(&bytes)
+            .context("invalid number of removed currency pairs bytes")?;
+        Ok(num_removed_currency_pairs)
+    }
+
+    #[instrument(skip(self))]
+    async fn get_currency_pair_state(
+        &self,
+        currency_pair: &CurrencyPair,
+    ) -> Result<Option<CurrencyPairState>> {
+        let bytes = self
+            .get_raw(&currency_pair_state_storage_key(currency_pair))
+            .await
+            .context("failed to get currency pair state from state")?;
+        match bytes {
+            Some(bytes) => {
+                let currency_pair_state = serde_json::from_slice(&bytes)
+                    .context("failed to deserialize currency pair state")?;
+                Ok(Some(currency_pair_state))
+            }
+            None => Ok(None),
+        }
+    }
+
+    #[instrument(skip(self))]
+    async fn get_all_currency_pairs(&self) -> Result<Vec<CurrencyPair>> {
+        let prefix = format!("{CURRENCY_PAIR_STATE_PREFIX}/");
+        let mut currency_pairs: Vec<CurrencyPair> = Vec::new();
+
+        let mut stream = std::pin::pin!(self.prefix_keys(&prefix));
+        while let Some(Ok(key)) = stream.next().await {
+            let currency_pair = key
+                .strip_prefix(&prefix)
+                .context("failed to strip prefix from currency pair state key")?
+                .parse::<CurrencyPair>()
+                .context("failed to parse storage key suffix as currency pair")?;
+            currency_pairs.push(currency_pair);
+        }
+        Ok(currency_pairs)
+    }
+
+    #[instrument(skip(self))]
+    async fn get_next_currency_pair_id(&self) -> Result<u64> {
```

Comment:
This method doesn't seem to be used anywhere.

Reply:
not currently but will be used when the slinky actions are implemented, should i remove for now?

Reply:
IMO yes so that once it's used it can be understood with the actions.